Add Copy elisions

Magnus-Mage opened the pull request:
Changes made:

	* wiki/resources/cpp/: added Copy_elisions.md

Sign off: Magnus-Mage <>

Magnus-Mage committed:
Add Copy elisions

Changes made:

	* wiki/resources/cpp/:
	added Copy_elisions.md

Sign off: Magnus-Mage <>

ProfessionalMenace reviewed (changes requested):
Change name from Copy_elisions.md to copy-elisions.md using
```
git mv Copy_elisions.md copy-elisions.md
```

Article not discoverable using a sidebar add
```
{
    text: "Copy Elision",
    link: "/resources/cpp/copy-elisions",
},
```

Magnus-Mage committed:
Add copy elision to sidebar

Changes made:

	* wiki/resources/cpp/Copy_elisions.md:
	Fixed the markdown issues.
	* wiki/resources/sidebar.ts
	added Copy_elisions under C++ resources

Signed off: Magnus-Mage <>

jeremy-rifkin commented on wiki/resources/cpp/Copy_elisions.md:
```diff
@@ -0,0 +1,733 @@
+
+## A Brief History
+
+Let's start with some context.
```
Best to avoid narration during wiki articles

jeremy-rifkin commented on wiki/resources/cpp/Copy_elisions.md:
```diff
@@ -0,0 +1,733 @@
+
+Let's start with some context.
+
+Back in 1988, compiler developers realized that programs were doing a lot of unnecessary copying when returning objects
```
Maybe include this history section as an aside somewhere? We'd like these wiki articles to be relatively focused.

jeremy-rifkin commented on wiki/resources/cpp/Copy_elisions.md:
```diff
@@ -0,0 +1,733 @@
+required to.
+
+The introduction of move semantics in C++11 helped reduce copying costs, but copy elision remained even better because
+it could eliminate operations entirely rather than just making them cheaper.
```
If going down the route of describing this stuff it would be worth noting that often copy elision and a move constructor invocation can be optimized the same thing, but copy elision bypasses move constructors all together which is helpful if the move constructor doesn't optimize away completely.

jeremy-rifkin reviewed (commented):
Thank you so much for taking the time to put this article together! I've left some comments below, happy to chat about things more

Magnus-Mage committed:
Update wiki/resources/cpp/Copy_elisions.md

Co-authored-by: Jeremy Rifkin <51220084+jeremy-rifkin@users.noreply.github.com>

Magnus-Mage committed:
Rename and revised copy elision article

- Rename Copy_elisions.md to copy-elision.md
- Remove unnecessary RVO hierarchy
- Condense history section into collapsible aside
- Remove narration and conversational elements
- Changed Headers for parameter and exception sections

jeremy-rifkin commented on wiki/resources/cpp/copy-elision.md:
```diff
@@ -0,0 +1,714 @@
+performance costs. Starting with C++17 some forms of copy elision are
+**guaranteed** by the language standard.
+
+::: info What This Means for You 
```
This block and other still need updates for the auto-formatter

Magnus-Mage committed:
Format changes

Magnus-Mage committed:
Merge branch 'main' into add/copy_elisions
